core/src/main/scala/org/apache/spark/api/python/SerDeUtil.scala

```diff
@@ -186,6 +186,9 @@ private[spark] object SerDeUtil extends Logging {
       val unpickle = new Unpickler
       iter.flatMap { row =>
         val obj = unpickle.loads(row)
+        // `Opcodes.MEMOIZE` of Protocol 4 (Python 3.4+) will store objects in internal map
+        // of `Unpickler`. This map is cleared when calling `Unpickler.close()`.
+        unpickle.close()
```

Comment:
I don't know anything about this, but it looks odd to close the object repeatedly. It may not cause a problem now. What's the downside to using a new object for each row, just performance?

Reply:
Yea, I missed `Since each row is independent of the others` - so I thought we should manually control `memo`. I had to look deeper :D.  Looks fine. Yes, I guess just it needs an extra clear call for each row.

Reply:
To use use a new object for each row is also ok. I didn't do it just for (possible) performance concern.